libqpdf/QPDFWriter.cc

```diff
@@ -1469,16 +1467,11 @@ QPDFWriter::willFilterStream(QPDFObjectHandle stream,
                              PointerHolder<Buffer>* stream_data)
 {
     compress_stream = false;
-    is_metadata = false;
 
     QPDFObjGen old_og = stream.getObjGen();
     QPDFObjectHandle stream_dict = stream.getDict();
 
-    if (stream_dict.getKey("/Type").isName() &&
-        (stream_dict.getKey("/Type").getName() == "/Metadata"))
-    {
-        is_metadata = true;
-    }
+    is_metadata = stream_dict.isDictionaryOfType("/Metadata");
```

Comment:
You can replace the if, but don't assign to is_metadata. Set is_metadata inside the if.

Probably best not to replace `if (condition) { a = true; }` with `a = condition` unless it's blindingly obvious like there are no intervening statements between the initial declaration and the assignment, but even then, sometimes it's the way it is for consistency.

Reply:
Done the change, although I don't necessarily agree.

To my mind, within the method `is_metadata` is a const bool defined by the assignment, and immutable from this point on. To me, `if (condition) { a = true; }` implies `a` is mutable and may be changed later in the method.

Reply:
That may be true in this case, but I frequently code in a way where I check many conditions and flip values of booleans as I go, and, in the absence of an `else`, you have to study the whole function to know whether the two are equivalent.